convert/float16/float16.go

```diff
@@ -0,0 +1,97 @@
+package float16
+
+import (
+	"math"
+)
+
+func FromFloat32s(f32s []float32) (u16s []uint16) {
+	u16s = make([]uint16, len(f32s))
+	for i := range f32s {
+		bits := math.Float32bits(f32s[i])
+		sign := (bits >> 31) & 0x1
```

Comment:
you could save the shift here by doing `bits & 0x8000000` and then shifting the sign back over when you're adding it back below (i.e. `u16s[i] = uint16((sign >> 16 ) | 0x7c00)`. Dunno how much time that would really save though.